test/Constraints/diag_ambiguities.swift

```diff
@@ -29,11 +29,12 @@ C(g) // expected-error{{ambiguous use of 'g'}}
 func h<T>(_ x: T) -> () {}
 C(h) // expected-error{{ambiguous use of 'init'}}
 
-func rdar29691909_callee(_ o: AnyObject?) -> Any? { return o } // expected-note {{found this candidate}}
-func rdar29691909_callee(_ o: AnyObject) -> Any { return o } // expected-note {{found this candidate}}
+func rdar29691909_callee(_ o: AnyObject?) -> Int? { fatalError() }
+func rdar29691909_callee(_ o: AnyObject) -> Int { fatalError() }
 
-func rdar29691909(o: AnyObject) -> Any? {
-  return rdar29691909_callee(o) // expected-error{{ambiguous use of 'rdar29691909_callee'}}
+func rdar29691909(o: AnyObject) -> Int {
+  let r = rdar29691909_callee(o)
+  return r
```

Comment:
This is a different test. Please leave the original one intact, where there's a same-type constraint.